tracer/contrib/database/sqltraced/utils.go

```diff
@@ -0,0 +1,136 @@
+package sqltraced
+
+import (
+	"database/sql/driver"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/go-sql-driver/mysql"
+	"github.com/lib/pq"
+	"golang.org/x/net/context"
+)
+
+// Return all information passed through the DSN:
+func parseDSN(driver interface{}, dsn string) (meta map[string]string, err error) {
+	switch driver.(type) {
+	case *pq.Driver:
+		meta, err = parsePostgresDSN(dsn)
+	case *mysql.MySQLDriver:
+		meta, err = parseMySQLDSN(dsn)
+	}
+	meta = normalize(meta)
+	return
```

Comment:
in general, this style of programming that uses implicit returns is a bit harder to read IMO. Just do the simple thing

```
var meta map[string]string
var err error
...
return meta, err
```